src/composables/useDocumentTitle.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { computed, ref, watch } from 'vue'
+import type { ComputedRef } from 'vue'
+import type { Route } from 'vue-router'
+
+import { t } from '@nextcloud/l10n'
+
+import { useDocumentVisibility } from './useDocumentVisibility.ts'
+import { useStore } from './useStore.js'
+import Router from '../router/router.js'
+import { EventBus } from '../services/EventBus.ts'
+import type { Conversation } from '../types/index.ts'
+
+/**
+ * Composable to check whether the page is visible.
+ */
+export function useDocumentTitle() {
+	const store = useStore()
+	const isDocumentVisible = useDocumentVisibility()
+
+	const defaultPageTitle = ref<string>(getDefaultPageTitle())
+	const savedLastMessageMap = ref<Record<string, number>>({})
+
+	const conversationList = computed(() => store.getters.conversationsList)
+	const actorId = computed(() => store.getters.getActorId())
+	const actorType = computed(() => store.getters.getActorType())
+
+	watch(conversationList, (newValue) => {
+		if (isDocumentVisible.value || document.title.startsWith('* ')) {
+			return
+		}
+
+		const newLastMessageMap = getLastMessageMap(newValue)
+		/**
+		 * @return {boolean} Returns true, if
+		 * - a conversation is newly added to lastMessageMap
+		 * - a conversation has a different last message id then previously
+		 */
+		const shouldShowAsterisk = Object.keys(newLastMessageMap).some(token => {
+			return !savedLastMessageMap.value[token] // Conversation is new
+				|| (savedLastMessageMap.value[token] !== newLastMessageMap[token] // Last message changed
+					&& newLastMessageMap[token] !== -1) // But is not from the current user
```

Comment:
Should we add if it's also visible ?

Reply:
If by visible you mean isDocumentVisible, I'd say no, as it wasn't the case before. Asterisk was shown on background tabs, to indicate some thing's changed

Reply:
No, I mean the new message will be visible to the user.